agent/tool-scripts/bpftrace

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+# -*- mode: shell-script; indent-tabs-mode: t; sh-basic-offset: 8; sh-indentation: 8; sh-indent-for-case-alt: + -*-
+
+# Example usage: pbench-register-tool --name=bpftrace -- --script=sample_prog.bt
+# Which becomes, after processing: /usr/bin/bpftrace sample_prog.bt
+
+script_path=$(dirname $0)
+script_name=$(basename $0)
+pbench_bin="$(
+	cd ${script_path}/..
+	/bin/pwd
+)"
+
+# source the base script
+. "$pbench_bin"/base
+
+# Perftool scripts must provide the following functions
+# 1) Install the tool
+# 2) Start data collection
+# 3) Stop data collection
+# 4) post-process the data
+
+# Defaults
+tool=bpftrace
+tool_bin=/usr/bin/$tool
+group=default
+dir="/tmp"
+script_present=0
+
+opts=$(getopt -q -o dgi --longoptions "dir:,script:,group:,iteration:,start,stop,install,postprocess" -n "getopt.sh" -- "$@")
+if [ $? -ne 0 ]; then
+	printf "\n"
+	printf "$script_name: you specified an invalid option\n\n"
+	printf "The following options are available: \n\n"
+	printf -- "\t--install,			install this perf tool\n"
+	printf "\n"
+	printf -- "\t--start|stop|post-process	start/stop/post-process the data collection\n"
+	printf -- "\t--iteration=int 		the iteration (required)\n"
+	printf -- "\t--group=str		the perftool group (required)\n"
+	printf -- "\t--dir=str			directory to store data collection (required)\n"
+	printf -- "\t--script=str		path to bpftrace script\n"
+	exit 1
+fi
+eval set -- "$opts"
+while true; do
+	case "$1" in
+	--install)
+		mode="install"
+		shift
+		;;
+	--start)
+		mode="start"
+		shift
+		;;
+	--stop)
+		mode="stop"
+		shift
+		;;
+	--postprocess)
+		mode="postprocess"
+		shift
+		;;
+	-d | --dir)
+		shift
+		if [ -n "$1" ]; then
+			dir="$1"
+			shift
+		fi
+		;;
+	-g | --group)
+		shift
+		if [ -n "$1" ]; then
+			group="$1"
+			shift
+		fi
+		;;
+	-i | --iteration)
+		shift
+		if [ -n "$1" ]; then
+			iteration="$1"
+			shift
+		fi
+		;;
+	--script)
+		shift
+		script_present=1
+		if [ -n "$1" ]; then
+			script=$(echo $1 | sed 's/^"\(.*\)"$/\1/')
+			shift
+		else
+			echo "you did not provide a script"
```

Comment:
Seems like we don't need this `else` branch now that you added the check at line 102.

Reply:
I think we need this, There are two cases: 1) where `--script=` is supplied with no value 2) and `--script` option is not provided at all. 
So:
1) `pbench-register-tool --name=bpftrace -- --script=`
2) `pbench-register-tool --name=bpftrace --`

Reply:
You are correct, @sourabhtk37, but other arguments need the same checking.  So rather than just checking for `--script` I have posted a PR against your branch in your git repo that suggests a number of changes to enhance the command line processing, error handling, and logic of what is required when and what is optional.